code/modules/clothing/masks/gasmask.dm

```diff
@@ -2,25 +2,76 @@
 	name = "gas mask"
 	desc = "A face-covering mask that can be connected to an air supply. While good for concealing your identity, it isn't good for blocking gas flow." //More accurate
 	icon_state = "gas_alt"
-	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS | GAS_FILTERING
 	flags_inv = HIDEEARS|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
 	w_class = WEIGHT_CLASS_NORMAL
 	inhand_icon_state = "gas_alt"
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH | PEPPERPROOF
 	resistance_flags = NONE
+	var/max_filters = 1
+	var/current_filters = 0
+	var/list/gas_filters
```

Comment:
Going along with what I said below, I'd make this more modular, to allow for interesting nonsense, down the road.

Why? Because, hypothetically, maybe we want to make a mask that filters or has other behavior; maybe we want to have a mask that *generates* some type of gas that's breathed in, or modifies the `breath` gas mixture in some way that we want to interact with.

Reply:
your first comment should be easy enough to make, i was thinking about it, but first i wanted to see if this was going through.
for this one, it could be more difficult, i'll see what can be done